app/mutations/useErrorMutation.ts

```diff
@@ -0,0 +1,52 @@
+// import { useContext } from "react";
+
+// const useMutation = (errorMessage) => {
+//   //on error
+//   //alert sentry
+//   //set the context to the errorMessage
+// };
+
+// export default useMutation;
+
+import { ErrorContext } from "contexts/ErrorContext";
+// import { CacheConfigWithDebounce } from "lib/relay/debounceMutationMiddleware";
+import { useContext } from "react";
+import { useMutation } from "react-relay";
+import { commitMutation as baseCommitMutation } from "relay-runtime";
+import {
+  Disposable,
+  GraphQLTaggedNode,
+  IEnvironment,
+  MutationConfig,
+  MutationParameters,
+} from "relay-runtime";
+import * as Sentry from "@sentry/nextjs";
+
+export default function useErrorMutation<TMutation extends MutationParameters>(
+  mutation: GraphQLTaggedNode,
+  errorMessage: string
```

Comment:
we might want a `getErrorMessage: (relayError: Error) => string`, to allow different error messages depending on the actual error message. The documentation should say something about this function returning a user-friendly error message.

Reply:
Would this be a parameter in addition to `errorMessage`, or a replacement for it? Also, is `getErrorMessage` a function that comes with relay, or am I supposed to write it? I couldn't import it or find it in the docs (I found a bit about [user-friendly messages in schemas](https://relay.dev/docs/guided-tour/rendering/error-states/#accessing-errors-in-graphql-responses), but I don't think that's the same thing).

Reply:
A replacement for `errorMessage`, which you would have to write. This would allow to display different alerts for a given mutation, depending on the original error. With a string `errorMessage`, your mutation will always end up displaying the same error if it fails.

This is more of a future need, since we haven't discussed the actual error messages yet. I would imagine that our first `getErrorMessage` functions will be something like `() => "An error happened while saving the form"`, ignoring the actual error.